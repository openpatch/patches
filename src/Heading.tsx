import { ReactNode } from "react";
import { Text, TextProps } from "./Text";

export type HeadingProps = {
  children?: ReactNode;
  as?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
} & TextProps;

export const Heading = ({ as = "h1", ...props }: HeadingProps) => {
  return <Text as={as} {...props} />;
};
