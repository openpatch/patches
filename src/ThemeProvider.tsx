import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
