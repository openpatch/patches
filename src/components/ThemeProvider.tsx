import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
