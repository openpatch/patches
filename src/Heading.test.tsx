import React from "react";
import { render } from "@testing-library/react";
import { Heading, HeadingProps } from "./Heading";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: HeadingProps) =>
  render(
    <ThemeProvider>
      <Heading {...props} />
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
