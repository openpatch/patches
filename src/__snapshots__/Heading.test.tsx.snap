// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  padding: 0px;
  margin: 0px;
}

<div>
  <div
    id="patches-root"
  >
    <h1
      class="emotion-0"
    />
  </div>
</div>
`;
